### PR TITLE
<FocusableHOC/> - don't hoist bind,apply and call methods

### PR DESCRIPTION
**Pull request by mykas**
This PR fixes warnings in wix-ui-backoffice and wix-style-react where is said that bind,apply and call methods should not be hoisted. Relevant component `Badge`. Warning happens because it is wrapped with FocusableHOC. Going back to using hoist-non-react-methods instead of hoist-non-react-statics. Also made sure that stateless components don't pass any refs and just passes required props.

### CODE REVIEW

**Comment by argshook** [comment]
what's the issue?

**Comment by mykas** [comment]
Well, we get this warning in wix-style-react and it seems that this is related with prototype methods being hoisted so I am not sure if this is ok to ignore them.

File: packages/wix-ui-core/src/hocs/Focusable/FocusableHOC.tsx [hunk 1]
```diff
@@ -98,8 +103,23 @@ export const withFocusable = Component => {
     };
 
     render() {
-      return (
+      return isStatelessComponent(Component) ? (
+        <Component
```
[the remark this answers is not in the document]
Comment (mykas) [reply packages/wix-ui-core/src/hocs/Focusable/FocusableHOC.tsx]: ok ok. But regarding hoist-non-react-statics I am not sure why its not handling properly prototype methods. But what I have noticed in hoist-non-react-methods is that it handles more methods then `hoist-non-react-statics`.

`hoist-non-react-methods`
const REACT_PROTOTYPE = {
  autobind: true,
  childContextTypes: true,
  componentDidMount: true,
  componentDidUpdate: true,
  componentWillMount: true,
  componentWillReceiveProps: true,
  componentWillUnmount: true,
  componentWillUpdate: true,
  contextTypes: true,
  displayName: true,
  forceUpdate: true,
  getChildContext: true,
  getDefaultProps: true,
  getDOMNode: true,
  getInitialState: true,
  isMounted: true,
  mixins: true,
  propTypes: true,
  render: true,
  replaceProps: true,
  replaceState: true,
  setProps: true,
  setState: true,
  shouldComponentUpdate: true,
  statics: true,
  updateComponent: true,
} 

`hoist-non-react-statics`
const REACT_STATICS = {
    childContextTypes: true,
    contextType: true,
    contextTypes: true,
    defaultProps: true,
    displayName: true,
    getDefaultProps: true,
    getDerivedStateFromProps: true,
    mixins: true,
    propTypes: true,
    type: true
};

**Comment by mykas** [comment]
Ready for review.

**Pull request merged**